src/pg_ext/big_decimal.cr

```diff
@@ -0,0 +1,21 @@
+require "big"
+
+module PG
+  struct Numeric
+    # Returns a BigDecimal representation of the numeric. This retains all precision.
+    def to_big_d
+      return BigDecimal.new("0") if nan? || ndigits == 0
```

Comment:
Just noticed that this should be `BigDecimal.new(0)` to avoid unnecessary allocation...